zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/MultiResourceDeploymentTest.java

```diff
@@ -0,0 +1,341 @@
+              .withXmlResource(PROCESS_V1)
+              .withXmlClasspathResource(DMN_V1)
+              .withJsonClasspathResource(FORM_V1)
+              .deploy()
+              .getValue();
+      final var processV1 = firstDeployment.getProcessesMetadata().getFirst();
+      final var drgV1 = firstDeployment.getDecisionRequirementsMetadata().getFirst();
+      final var decisionV1 = firstDeployment.getDecisionsMetadata().getFirst();
+      final var formV1 = firstDeployment.getFormMetadata().getFirst();
+
+      // when
+      final var secondDeployment =
+          engine
+              .deployment()
+              // deploy the exact same resources again
+              .withXmlResource(PROCESS_V1)
+              .withXmlClasspathResource(DMN_V1)
+              .withJsonClasspathResource(FORM_V1)
+              .deploy()
+              .getValue();
+
+      // then
+      assertThat(secondDeployment.getProcessesMetadata())
+          .singleElement()
+          .satisfies(
+              metadata ->
+                  Assertions.assertThat(metadata)
+                      .hasVersion(1)
+                      .isDuplicate()
+                      .extracting(
+                          ProcessMetadataValue::getProcessDefinitionKey,
+                          InstanceOfAssertFactories.LONG)
+                      .isEqualTo(processV1.getProcessDefinitionKey()));
+      assertThat(secondDeployment.getDecisionRequirementsMetadata())
+          .singleElement()
+          .satisfies(
+              metadata ->
+                  Assertions.assertThat(metadata)
+                      .hasDecisionRequirementsVersion(1)
+                      .isDuplicate()
+                      .extracting(
+                          DecisionRequirementsMetadataValue::getDecisionRequirementsKey,
+                          InstanceOfAssertFactories.LONG)
+                      .isEqualTo(drgV1.getDecisionRequirementsKey()));
+      assertThat(secondDeployment.getDecisionsMetadata())
+          .singleElement()
+          .satisfies(
+              metadata ->
+                  Assertions.assertThat(metadata)
+                      .hasVersion(1)
+                      .isDuplicate()
+                      .extracting(
+                          DecisionRecordValue::getDecisionKey,
+                          DecisionRecordValue::getDecisionRequirementsKey)
+                      .containsExactly(
+                          decisionV1.getDecisionKey(), drgV1.getDecisionRequirementsKey()));
+      assertThat(secondDeployment.getFormMetadata())
+          .singleElement()
+          .satisfies(
+              metadata ->
+                  Assertions.assertThat(metadata)
+                      .hasVersion(1)
+                      .isDuplicate()
+                      .extracting(FormMetadataValue::getFormKey)
+                      .isEqualTo(formV1.getFormKey()));
+
+      assertThat(RecordingExporter.processRecords().withIntent(ProcessIntent.CREATED).limit(2))
```

Comment:
I wonder why do we have limit 2 here? This is what adds time to the test execution. The second record is never found, so we keep waiting for 5 seconds until timeout is reached. Same applies for the below assertions.

Reply:
Hi @ana-vinogradova-camunda,
I wanted to ensure that no second record is written unintentionally. If there was a second one, the `containsExactly` assertion would fail. I think I could use `limit(1)` to avoid the 5 seconds waiting time, but then I probably wouldn't know for sure if a second record was written.

Reply:
Maybe that's also fine. I'm not sure if there is a very good way to test that a record was _not_ written ;)

Reply:
Thanks @mustafadagher for the hint! I've refactored the test now a bit and was able to reduce the run time to ~7 seconds while still ensuring that no new records have been written. I hope that this is a good compromise 🙂.

Reply:
@jfriedenstab it runs in 12 seconds on my machine.